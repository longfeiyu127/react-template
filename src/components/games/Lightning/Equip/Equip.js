import React from 'react';
// import Tictactoe from './tictactoe'
// import Fivechess from './fivechess'
import './Equip.less';

class Equip extends React.Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    // eslint-disable-next-line
    const { life, bomb } = this.props;
    return (
      <div className="g-lig-equip">
        <div className="content">
          <div className="icon life" />
          <p className="count">{life}</p>
        </div>
        <div className="content">
          <div className="icon bomb" />
          <p className="count">{bomb}</p>
        </div>
      </div>
    );
  }
}

export default Equip;
