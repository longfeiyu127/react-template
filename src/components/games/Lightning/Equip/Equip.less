@import '../../../../assets/styles/theme.less';
@font-face {font-family: "game";
  src: url('../../../../assets/font/game.ttf') format('truetype');
}
.g-lig-equip{
  position: absolute;
  left: 0.1rem;
  bottom: 0.1rem;
  width: 0.7rem;
  height: 1.13rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  .content{
    flex: 1;
    display: flex;
    justify-content: space-between;
    align-items: center;
    >.icon{
      width: 0.45rem;
      height: 0.45rem;
    }
    >.life {
      background: url('../../../../assets/images/games/lightning/equip/life_count.png') no-repeat center/100% 100%;
    }
    >.bomb{
      background: url('../../../../assets/images/games/lightning/equip/bomb_count.png') no-repeat center/100% 100%;
    }
    >.count{
      color: #ff3314;
      font-size: 0.3rem;
      font-family: game;
      .count();
      font-weight: 800;
    }
  }
  // >.player{
  //   background: url('../../../../../assets/images/games/lightning/bullet/palyerBullet.png') no-repeat center/100% 100%;
  //   animation: flicker 100ms infinite;
  // }
  // >.power{
  //   background: url('../../../../../assets/images/games/lightning/materials/power_up_01.png') no-repeat center/100% 100%;
  // }

  // @keyframes flicker {
  //   0%, 100% {
  //     opacity: 0;
  //   }
  //   50% {
  //     opacity: 1;
  //   }
  // }
}
