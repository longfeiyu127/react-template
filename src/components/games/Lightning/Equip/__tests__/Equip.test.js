import React from 'react';
import renderer from 'react-test-renderer';
import Equip from '../Equip';

describe('<Equip />', () => {
  it('component render', () => {
    let equip = renderer.create(<Equip life={1} bomb={1} />).toJSON();
    expect(equip).toMatchSnapshot();

    equip = renderer.create(<Equip life={0} bomb={1} />).toJSON();
    expect(equip).toMatchSnapshot();
  });
});
