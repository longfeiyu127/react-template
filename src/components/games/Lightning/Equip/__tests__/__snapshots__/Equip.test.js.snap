// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Equip /> component render 1`] = `
<div
  className="g-lig-equip"
>
  <div
    className="content"
  >
    <div
      className="icon life"
    />
    <p
      className="count"
    >
      1
    </p>
  </div>
  <div
    className="content"
  >
    <div
      className="icon bomb"
    />
    <p
      className="count"
    >
      1
    </p>
  </div>
</div>
`;

exports[`<Equip /> component render 2`] = `
<div
  className="g-lig-equip"
>
  <div
    className="content"
  >
    <div
      className="icon life"
    />
    <p
      className="count"
    >
      0
    </p>
  </div>
  <div
    className="content"
  >
    <div
      className="icon bomb"
    />
    <p
      className="count"
    >
      1
    </p>
  </div>
</div>
`;
